hw_9_decorators/pack_9_dekorators/task_1.py

```diff
@@ -0,0 +1,92 @@
+# Напишите следующие функции:
+# * Нахождение корней квадратного уравнения
+# * Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+# * Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+# * Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+import csv
+import math
+import random
+import json
+from functools import wraps
+
+# Нахождение корней квадратного уравнения
+def solve_quadratic_equation(a, b, c):
+    discriminant = b ** 2 - 4 * a * c
+    if discriminant < 0:
+        return None
+    elif discriminant == 0:
+        x = -b / (2 * a)
+        return x
+    else:
+        x1 = (-b + math.sqrt(discriminant)) / (2 * a)
+        x2 = (-b - math.sqrt(discriminant)) / (2 * a)
+        return x1, x2
+
+# Генерация csv файла с тремя случайными числами в каждой строке (100-1000 строк)
+def generate_csv_file(file_name, rows):
+    with open(file_name, 'w', newline='') as file:
+        writer = csv.writer(file)
+        for _ in range(rows):
+            numbers = [random.randint(1, 1000) for _ in range(3)]
+            writer.writerow(numbers)
+
+# Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла
+def solve_quadratic_equation_from_csv(func):
+    @wraps(func)
+    def wrapper(file_name):
+        with open(file_name, 'r') as file:
+            reader = csv.reader(file)
+            for row in reader:
+                numbers = list(map(int, row))
+                result = func(*numbers)
+                print(f"For numbers {numbers}, quadratic equation roots: {result}")
+    return wrapper
+
+# Декоратор, сохраняющий переданные параметры из csv файла в функцию нахождения корней квадратного уравнения
+# и результаты работы функции в json файл
+def save_to_json(func):
+    func_name = func.__name__
+
+    def decorator(file_name):
+        result_list = []
+        with open(file_name, 'r') as file:
+            reader = csv.reader(file)
+            for row in reader:
+                numbers = list(map(int, row))
+                result = func(*numbers)
+                result_dict = {
+                    'args': numbers,
+                    'result': result
+                }
+                result_list.append(result_dict)
```

Comment:
Что-то я не понял. Ведь этот код повторяется в предыдущем декораторе. Зачем он тут?
```suggestion
    result_list = []
    def decorator(*args):
        result = func(*numbers)
        result_dict = {
            'args': numbers,
            'result': result
        }
        result_list.append(result_dict)
```